Branch-SDK/src/main/java/io/branch/referral/validators/IntegrationValidator.java

```diff
@@ -217,7 +217,7 @@ private boolean checkIfIntentAddedForLinkDomain(String domainName) {
         boolean foundIntentFilterMatchingDomainName = false;
         if (!TextUtils.isEmpty(domainName) && integrationModel.applinkScheme != null) {
             for (String host : integrationModel.applinkScheme) {
-                if (domainName.equals(host)) {
+                if (domainName.equals(integrationModel.getStringResource(host))) {
```

Comment:
At this point we don't know if `host` is a string resource or the actual string, so we should either rename the method to something like `maybeExtractStringResource` or add another util method to test if it is indeed a string resource and only then pass it your new method `getStringResource`.